### ISSUE TITLE
Running disown inside fish_postexec creates zombie process

### ISSUE THREAD

**Issue by afreakk** [fish-shell/fish-shell#7627]
Fish --version: 3.2.1
OS: NixOS, kernel 5.4.87

If I define a function like this:
```
function __event_test --on-event="fish_postexec"
   sleep 2 &; disown
end
```
and keep an eye on `htop`, the `sleep` processes never dies, they become zombies.
(The same goes for the other processes Ive tried, like I tried running a program that syncs my fish-history with other computers, but I want to have it run in the background until exit, but it never exits)

**Comment by faho** [comment]
Duplicate of #7183

**Comment by faho** [comment]
(and also #7581), which was fixed by #7189

**Comment by afreakk** [comment]
aha, thanks, will wait for 3.1.3 then :+1:
